### PR TITLE
Ee 6825 component trace - happy path only

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
- Added handling of a x-component-trace header to the response. 
- When the component receives a request it adds itself to the header.
- When the component calls hmrc-service it replaces the header with that returned by hmrc-service. If no header is recieved then the current one is retained.
- When this component responds to the caller, it includes this new header.

I have introduced a new pattern, using a ResponseBodyAdvice component to add the header to all responses. I couldn't do this in RequestData because by the time the postHandle and afterCompletion methods are called, the response has been committed (so new headers are ignored) - see https://mtyurt.net/post/spring-modify-response-headers-after-processing.html

I have only done the "happy-path" adding the component trace from hmrc-service if it returns a successful response so that we can assess the pattern before I do audit-service and non-happy-path.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/proving/income/api/RequestData.java [hunk 1]
```diff
@@ -62,6 +65,14 @@ private String initialiseRequestStart() {
         return Long.toString(requestStart);
     }
 
+    private String initialiseComponentTrace(HttpServletRequest request) {
+        String componentTrace = request.getHeader(COMPONENT_TRACE_HEADER);
+        if (componentTrace == null) {
+            return "pttg-ip-api";
```
Comment (Tim-Hammonds) [inline src/main/java/uk/gov/digital/ho/proving/income/api/RequestData.java]: Constant here

File: src/test/groovy/uk/gov/digital/ho/proving/income/api/test/FinancialServiceSpec.groovy [hunk 2]
```diff
@@ -39,13 +39,17 @@ class FinancialServiceSpec extends Specification {
     def mockAuditClient = Mock(AuditClient)
     def mockNinoUtils = Mock(NinoUtils)
     def mockIncomeValidationService = Mock(IncomeValidationService)
-    def mockRequestData = Mock(RequestData)
+    def mockRequestData = Stub(RequestData)
```
Comment (Tim-Hammonds) [inline src/test/groovy/uk/gov/digital/ho/proving/income/api/test/FinancialServiceSpec.groovy]: Rename a stubRequestData

**Review by Tim-Hammonds — APPROVED** [review]

**Comment by owainrobertsbjss** [comment]
Note to self: Also disussed moving the code for adding the response header into RequestData (although the ControllerAdvice will still be the point that it is called)

**Comment by owainrobertsbjss** [comment]
> Note to self: Also disussed moving the code for adding the response header into RequestData (although the ControllerAdvice will still be the point that it is called)

This is also addressed now

**Pull request merged**